arrow-libs/fx/arrow-fx-coroutines/src/commonTest/kotlin/arrow/fx/coroutines/CyclicBarrierSpec.kt

```diff
@@ -21,54 +18,99 @@ class CyclicBarrierSpec : StringSpec({
         "Cyclic barrier must be constructed with positive non-zero capacity $i but was $i > 0"
     }
   }
-  
+
   "barrier of capacity 1 is a no op" {
     checkAll(Arb.constant(Unit)) {
       val barrier = CyclicBarrier(1)
       barrier.await()
     }
   }
-  
+
   "awaiting all in parallel resumes all coroutines" {
     checkAll(Arb.int(1, 100)) { i ->
       val barrier = CyclicBarrier(i)
-      (0 until i).parTraverse { barrier.await() }
+      (0 until i).parMap { barrier.await() }
     }
   }
-  
+
   "should reset once full" {
     checkAll(Arb.constant(Unit)) {
       val barrier = CyclicBarrier(2)
       parZip({ barrier.await() }, { barrier.await() }) { _, _ -> }
       barrier.capacity shouldBe 2
     }
   }
-  
+
+  "executes runnable once full" {
+    checkAll(Arb.constant(Unit)) {
+      var barrierRunnableInvoked = false
+      val barrier = CyclicBarrier(2) { barrierRunnableInvoked = true }
+      parZip({ barrier.await() }, { barrier.await() }) { _, _ -> }
+      barrier.capacity shouldBe 2
+      barrierRunnableInvoked shouldBe true
+    }
```

Comment:
No need to wrap in `checkAll` if you're not using the generator at all

```suggestion
    var barrierRunnableInvoked = false
    val barrier = CyclicBarrier(2) { barrierRunnableInvoked = true }
    parZip({ barrier.await() }, { barrier.await() }) { _, _ -> }
    barrier.capacity shouldBe 2
    barrierRunnableInvoked shouldBe true
```